sample/src/main/java/com/willowtreeapps/signinwithapple/sample/SampleActivity.kt

```diff
@@ -6,27 +6,33 @@ import android.widget.Toast
 import android.widget.Toast.LENGTH_SHORT
 import androidx.appcompat.app.AppCompatActivity
 import androidx.fragment.app.FragmentManager
-import com.willowtreeapps.signinwithapplebutton.SignInWithAppleClient
+import com.willowtreeapps.signinwithapplebutton.SignInWithAppleCallback
 import com.willowtreeapps.signinwithapplebutton.SignInWithAppleService
 import com.willowtreeapps.signinwithapplebutton.view.SignInWithAppleButton
 
-class SampleActivity : AppCompatActivity(), SignInWithAppleClient {
+class SampleActivity : AppCompatActivity() {
 
     private lateinit var signInWithAppleButtonBlack: SignInWithAppleButton
     private lateinit var signInWithAppleButtonWhite: SignInWithAppleButton
     private lateinit var signInWithAppleButtonWhiteOutline: SignInWithAppleButton
 
+    private val callback = object: SignInWithAppleCallback {
+        override fun onSignInWithAppleSuccess(authorizationCode: String) {
+            Toast.makeText(this@SampleActivity, authorizationCode, LENGTH_SHORT).show()
+        }
+
+        override fun onSignInWithAppleFailure(error: Throwable) {
+            Log.d("SAMPLE_APP", "Received error from Apple Sign In ${error.message}")
+        }
+    }
```

Comment:
I take it it's more idiomatic to conform in an anonymous object than in the caller? Pardon my iOS sensibilities.

Reply:
I'd say _it depends_, I'd say it's pretty common when dealing with activity/fragment as you want to decouple your stuff from those classes as much as possible. Wouldn't be against changing it back in the sample app though.

Reply:
Nah, if it's common and good practice then let's set a good example.